### PR TITLE
boards: add support for i-nucleo-lrwan1 (Arduino-like shield)

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This board provides LoRa connectivity via SX1272 radio and is running an STM32L052 MCU.
All [documentation is on GitHub](https://github.com/USIWP1Module/USI_I-NUCLEO-LRWAN1).

After closing R19/R20, the on-board sensors (LPS22HB, HTS221) can also be used.

To flash the board, an external programmer (ST-Link is configured by default) must be used.

I couldn't test LoRaWAN connectivity yet.

Things tested with success:
- RTC/RTT
- Timers
- I2C
- SPI

### Testing procedure

Run `compile_and_test_for_one_board.py` and it should succeed

### Issues/PRs references

None

### CODE REVIEW

**Comment by aabadie** [comment]
@fjmolinas, it works when using PA boost :)

File: boards/i-nucleo-lrwan1/doc.txt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
++            -c 'init' \
++            -c 'targets' \
++            -c 'reset halt' \
+             ${OPENOCD_PRE_FLASH_CMDS} \
+             -c 'flash write_image erase \"${IMAGE_FILE}\" ${IMAGE_OFFSET} ${IMAGE_TYPE}' \
+             ${OPENOCD_PRE_VERIFY_CMDS} \
+```
+- Run make flash:
+```sh
+make BOARD=i-nucleo-lrwan1 -C examples/hello-world flash
```
Comment (fjmolinas) [inline boards/i-nucleo-lrwan1/doc.txt]: Add a statement saying the flash will fail but the flash memory will be unlocked after power cycle (or plugging and unplugging as you stated)

**Comment by aabadie** [comment]
About the failing tests:
- tests/pkg_c25559 is working for me
- other tests are failing because of insufficient RAM I would say. Should I add the board in the corresponding variable in the Makefile application ?

File: boards/i-nucleo-lrwan1/doc.txt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+## Description
+
+This board provides LoRa connectivity with an SX1272 radio connected via SPI
+to an ultra-low power
+[STM32L052T8 microcontroller](https://www.st.com/en/microcontrollers-microprocessors/stm32l052t8.html).
+
+For details, the
+[schematics are available on GitHub](https://github.com/USIWP1Module/USI_I-NUCLEO-LRWAN1/blob/master/Schematics/USI%20LoRa%20Arduino%20shield_SCH_20161115-1.pdf).
+
+_Note_: to use the on-board I2C sensors, the R19 and R20 shorts must be closed
```
[the remark this answers is not in the document]
Comment (aabadie) [reply boards/i-nucleo-lrwan1/doc.txt]: There's already a link to the pdf on github. I could prepare an image of it and embed in the generated html but I think it's an extra maintenance burden.

**Comment by aabadie** [comment]
Comments addressed @fjmolinas !

**Pull request merged**